js/src/dapps/tokenreg/Inputs/Text/input-text.js

```diff
@@ -105,22 +105,22 @@ export default class InputText extends Component {
     const { validationType, contract } = this.props;
     const validation = validate(value, validationType, contract);
 
-    if (validation instanceof Promise) {
+    const loadingTimeout = setTimeout(() => {
       this.setState({ disabled: true, loading: true });
+    }, 50);
```

Comment:
Doing this always feels very hacky, i.e. you are working around something you shouldn't be doing :) Only a mustn't-grumble.